### ISSUE TITLE
Dryrun leads to TypeError

### ISSUE THREAD

**Issue by maeg12** [sphinx-contrib/confluencebuilder#592]
Dryrun leads to Exception:
```
 File "D:\workspace\tesla\phoenix\.venv\lib\site-packages\sphinxcontrib\confluencebuilder\publisher.py", line 920, in _dryrun
    s += ' ' + self._name_cache[id_]
TypeError: can only concatenate str (not "list") to str
```
Detailed Error Report:
[sphinx-err-pqr81l0t.log](https://github.com/sphinx-contrib/confluencebuilder/files/7625518/sphinx-err-pqr81l0t.log)

I guess, https://github.com/sphinx-contrib/confluencebuilder/blob/c7cf8d26e8073aa929f0d40cd46434a46acd775d/sphinxcontrib/confluencebuilder/publisher.py#L383 should be `self._name_cache[page_id] = page['title']` instead of `self._name_cache[page_id] = ['title']`

Report
```
(system)
 platform: Windows-10-10.0.19043-SP0
   python: 3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit (AMD64)]
   sphinx: 4.3.0
 requests: 2.26.0
  builder: 1.7.0

(configuration)
confluence_page_hierarchy: True
confluence_publish: True
confluence_publish_dryrun: True
confluence_publish_postfix:  <WIk-123>
confluence_publish_root: (set)
confluence_purge: True
confluence_server_pass: (set)
confluence_server_url: https://<removed> (cloud)
confluence_server_user: (set)
confluence_space_key: (set; upper)

(confluence instance)
 connected: yes
   fetched: yes
   decoded: yes
    parsed: yes
      type: confluence
   version: 1000.0.0-ea405cde1aae
     build: 6452
```

**Comment by jdknight** [comment]
@maeg12, thanks for reporting this.

Unfortunately, a lot of the publishing logic is not yet covered in the unit tests which led to this regression. Changes have been added in #593 to correct this issue. This change has been added into the main branch and should be made available next stable release (v1.8). For now, users wanting to use the `confluence_publish_dryrun` option can workaround this by installing the [development version](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation). I will also try to see if I can cut a v1.7.1 release with this fix.

**Comment by jdknight** [comment]
v1.7.1 is now available on PyPI -- marking as closed.

If the changes applied have not resolved your issue, please feel free to mention it.
